### ISSUE TITLE
Can't import Layer from tensorflow.python.keras.engine.base_layer

### ISSUE THREAD

**Issue by harshithdwivedi** [tensorflow/tensorflow#36814]
I'm getting an error that says `from tensorflow.python.keras.engine.base_layer import Layer` while importing tensorflow in my app.

**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): MacOS
- TensorFlow installed from (source or
binary): binary
- TensorFlow version (use command below): 1.15.0
- Python version: 3.7

**Describe the current behavior**
On packaging an app using tensorflow with py2app, I get the following error while running the app: 
```
File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/__init__.py", line 36, in <module>
    from tensorflow._api.v1 import compat
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/_api/v1/compat/__init__.py", line 23, in <module>
    from tensorflow._api.v1.compat import v1
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/_api/v1/compat/v1/__init__.py", line 40, in <module>
    from tensorflow._api.v1.compat.v1 import experimental
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/_api/v1/compat/v1/experimental/__init__.py", line 11, in <module>
    from tensorflow.python.ops.control_flow_v2_toggles import output_all_intermediates
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/ops/control_flow_v2_toggles.py", line 24, in <module>
    from tensorflow.python.ops import control_flow_util_v2
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/ops/control_flow_util_v2.py", line 28, in <module>
    from tensorflow.python.keras.engine import base_layer_utils
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/__init__.py", line 27, in <module>
    from tensorflow.python.keras import applications
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/applications/__init__.py", line 25, in <module>
    from tensorflow.python.keras import engine
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/engine/__init__.py", line 23, in <module>
    from tensorflow.python.keras.engine.base_layer import Layer
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/engine/base_layer.py", line 58, in <module>
    from tensorflow.python.keras.saving.saved_model import save as saved_model
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/saving/saved_model/save.py", line 30, in <module>
    from tensorflow.python.keras.saving import saving_utils
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/saving/__init__.py", line 33, in <module>
    from tensorflow.python.keras.saving.saved_model_experimental import export_saved_model
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/saving/saved_model_experimental.py", line 30, in <module>
    from tensorflow.python.keras.utils import mode_keys
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/utils/__init__.py", line 38, in <module>
    from tensorflow.python.keras.utils.multi_gpu_utils import multi_gpu_model
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/utils/multi_gpu_utils.py", line 22, in <module>
    from tensorflow.python.keras.engine.training import Model
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/engine/training.py", line 42, in <module>
    from tensorflow.python.keras import metrics as metrics_module
  File "/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/metrics.py", line 34, in <module>
    from tensorflow.python.keras.engine.base_layer import Layer
ImportError: cannot import name 'Layer' from 'tensorflow.python.keras.engine.base_layer' (/Users/harshitdwivedi/Desktop/aftershootscripts/dist/aftershoot.app/Contents/Resources/lib/python3.7/tensorflow_core/python/keras/engine/base_layer.py)
```

**Describe the expected behavior**
The packaged app should work as expected.

**Code to reproduce the issue** Provide a reproducible test case that is the
bare minimum necessary to generate the problem.

```
import tensorflow as tf

self.model_path = r"{}".format(
            pathlib.Path(os.getcwd()).joinpath('model'))
        self.session = tf.Session(graph=tf.Graph())
        tf.saved_model.loader.load(self.session, ['serve'], self.model_path)
```

here's the setup.py that I've used with py2app: 

```
"""
This is a setup.py script generated by py2applet

Usage:
    python setup.py py2app
"""

from setuptools import setup

APP = ['aftershoot.py']
DATA_FILES = []
OPTIONS = {
        'packages' : ['PySide2', 'shiboken2', 'tensorflow_core', 'astor'],
        'resources' : ['exposure/']
        # 'optimize' : 1,
        }

setup(
    app=APP,
    data_files=DATA_FILES,
    options={'py2app': OPTIONS},
    setup_requires=['py2app']
)
```

**Comment by amahendrakar** [comment]
@harshithdwivedi,
I tried to reproduce the error given above but was unable to do so, looks like the given code is incomplete. Please find the Gist of it [here](https://colab.sandbox.google.com/gist/amahendrakar/8aad3e50a6496ef82b2443dc629df54a/36814.ipynb).

In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here. Thanks!

**Comment by harshithdwivedi** [comment]
Can you try packaging the app with py2app and then try running the packaged app?
This problem doesn't arise with normally importing tensorflow

**Comment by gowthamkpr** [comment]
@harshithdwivedi Can you please give the detailed steps to package the app with py2app.Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by RohanUBagewadi** [comment]
yes the issue still exists.

**Comment by guramritpalsaggu** [comment]
Yes The issue is still there please try fixing it asap.

**Comment by harshithdwivedi** [comment]
This issue still exists.

**Comment by guramritpalsaggu** [comment]
This issue still persists.

Regards
Guramritpal Saggu

**Comment by shipleyxie** [comment]
I got the same Error while using _Pycharm conda env in WIN10_, but the Error will **disappear** in _anaconda prompt_.

Steps to reproduce the Error:

1. Open Anaconda Prompt :`conda create -n env_name tensorflow==1.15.0 python==3.7`

2. Open Pycharm choose exist conda enviroment "env_name"

3. Just open a file type 
```python
import tensorflow
print("It will got Error")
"""
ImportError: cannot import name 'Layer' from 'tensorflow.python.keras.engine.base_layer' (.../python3.7/tensorflow_core/python/keras/engine/base_layer.py)
"""
```
4. Error disappear in anaconda prompt while using `import tensorflow`

**Comment by luisvivasg** [comment]
This also happens to me.  I am using Imageai and I downgraded to version 1.14

**Comment by Animadversio** [comment]
This also happen to me when I import tensorflow in matlab api 
tensorflow version 1.15.0
```matlab
py.importlib.import_module("tensorflow")
```

> Error using metrics><module> (line 34)
Python Error: ImportError: cannot import name 'Layer' from 'tensorflow.python.keras.engine.base_layer'
(C:\Users\binxu\.conda\envs\tf\lib\site-packages\tensorflow_core\python\keras\engine\base_layer.py)

Although I can import tensorflow fine in python. in the same environment.

**Comment by luisvivasg** [comment]
> This also happen to me when I import tensorflow in matlab api on Windows.
> Not sure if this could provide some clue to the bug.....
> tensorflow version 1.15.0
> 
> ```matlab
> py.importlib.import_module("tensorflow")
> ```
> 
> > Error using metrics> (line 34)
> > Python Error: ImportError: cannot import name 'Layer' from 'tensorflow.python.keras.engine.base_layer'
> > (C:\Users\binxu.conda\envs\tf\lib\site-packages\tensorflow_core\python\keras\engine\base_layer.py)
> 
> Although I can import tensorflow fine in python. in the same environment.

I found that the problem occurs when I was downgrading Tensorflow to version 1.14. What I did was to keep version 2.+ then I did not find this error.

I needed to downgrade since everywhere it said that YOLO required Tensorflow 1.14. If that is your case you need to edit the YOLO code. everywhere it says import keras._whatever_, add tensorflow.keras._whatever_ 

If if it not your case and you have limitation with Tensorflow try upgrading it and you should not get this erro.

**Comment by Aarlington** [comment]
Locate `tensorflow_core/python/keras/metrics.py` and on line 34 change the import to `from tensorflow_core.python.keras.engine.base_layer import Layer` - you might have to do this for every subsequent error thrown due to the mismatched library names.

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.

**Comment by piaobuliao** [comment]
in my problem,I have two  Anaconda versions in PATH, I delete the other Anaconda path in my PATH(windows system) , Solve the problem
..
first my error info:
"ImportError: cannot import name 'Layer' from partially initialized module 'tensorflow.python.keras.engine.base_layer' (most likely due to a circular import) (D:\download\Anaconda3-2021.05-python3.8\lib\site-packages\tensorflow_core\python\keras\engine\base_layer.py)"


delete the other Anaconda path in my PATH(windows system) , Solve the problem
OK..
>>> import tensorflow
2021-08-28 20:07:35.466805: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudart64_110.dll
WARNING:tensorflow:Deprecation warnings have been disabled. Set TF_ENABLE_DEPRECATION_WARNINGS=1 to re-enable them.
>>>

**Comment by Sajan4072** [comment]
In my case I was trying to run tensorflow(2.0.0a0) in my django(3.0) app and while using tensorflow in the backend , I got the following error.
![error](https://user-images.githubusercontent.com/36479349/131254101-d5ef9a77-5488-4e6c-b242-54cdb8731953.PNG)

**Comment by otdtds** [comment]
I also have this problem.How do you solve this problem in last?

**Comment by JHCloud0115** [comment]
How did you solve the problem? I keep getting the same error

**Comment by Petros626** [comment]
@harshithdwivedi  solution found for it?

**Comment by harshithdwivedi** [comment]
@Petros626 i was using pyinstaller for packaging and shipping TensorFlow. Updating to the latest version of pyinstaller fixes the issue for TensorFlow v2.5.X

**Comment by pengzhuoyu** [comment]
I am using the latest version of tensorflow and still have such problem. My tensorflow is installed using anaconda
